mlir/lib/Dialect/EmitC/IR/EmitC.cpp

```diff
@@ -954,6 +974,10 @@ LogicalResult emitc::ArrayType::verify(
   for (int64_t dim : shape) {
     if (dim < 0)
       return emitError() << "dimensions must have non-negative size";
+    // TODO support dynamic dimensions on the outermost dimension,
+    // like int (*a)[50][100]
```

Comment:
Actually, the type of this is a pointer to array, and not an array with dynamic dimensions in C++. Maybe we represent it the same way in emitc (and thus remove this comment)